Source/WTF/wtf/text/ASCIILiteral.h

```diff
@@ -122,6 +123,12 @@ struct ASCIILiteralHash {
 template<typename T> struct DefaultHash;
 template<> struct DefaultHash<ASCIILiteral> : ASCIILiteralHash { };
 
+struct ASCIILiteralPtrHash {
+    static unsigned hash(const ASCIILiteral& key) { return IntHash<uintptr_t>::hash(reinterpret_cast<uintptr_t>(key.characters())); }
+    static bool equal(const ASCIILiteral& a, const ASCIILiteral& b) { return a.characters() == b.characters(); }
```

Comment:
Is it guaranteed that the linker will unique all ASCII literal pointers? Feels like it _might_ be, but I'm not sure.

Reply:
I don't think it does but this is not an issue here.

ASCIILiteral keeps using the other ASCIILiteralHash in general.

This new ASCIILiteralPtrHash is only for the SupplementMaps because they were already using PtrHash with `const char*`. Supplementable already required that keys (when they were of type `const char*`) were unique instances of a string literals.

Therefore, I carried over this behavior when switching over to ASCIILiteral to make sure I don't make perf worse.

Reply:
Maybe not guaranteed?

> Evaluating a [string-literal](https://timsong-cpp.github.io/cppwp/lex.string#nt:string-literal) results in a string literal object with static storage duration ([[basic.stc]](https://timsong-cpp.github.io/cppwp/basic.stc))[.](https://timsong-cpp.github.io/cppwp/lex.string#9.sentence-1) Whether all [string-literal](https://timsong-cpp.github.io/cppwp/lex.string#nt:string-literal)s are distinct (that is, are stored in nonoverlapping objects) and whether successive evaluations of a [string-literal](https://timsong-cpp.github.io/cppwp/lex.string#nt:string-literal) yield the same or a different object is unspecified[.](https://timsong-cpp.github.io/cppwp/lex.string#9.sentence-2)

https://timsong-cpp.github.io/cppwp/lex.string#9

Reply:
I did not change behavior AFAIK. The supplementable maps were already using PtrHash with `const char*`. All supplementable keys come from static functions on their respective classes.